ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Deployment/DeploymentPageVulnerabilities.tsx

```diff
@@ -142,6 +145,19 @@ function DeploymentPageVulnerabilities({ deploymentId }: DeploymentPageVulnerabi
     const vulnerabilityData = vulnerabilityRequest.data ?? vulnerabilityRequest.previousData;
     const totalVulnerabilityCount = vulnerabilityData?.deployment?.imageVulnerabilityCount ?? 0;
 
+    const deploymentNotFound =
+        (vulnerabilityData && !vulnerabilityData.deployment) ||
+        (summaryData && !summaryData.deployment);
+
+    if (deploymentNotFound) {
```

Comment:
This is needed here as well as in the parent `DeploymentPage` component because the top level component sends out all deployment related requests in parallel (in order to avoid the rest of the page blocking until each request succeeds).

The end result is the same, if any of the three requests returns `deployment: null`, the page will display a 404. This only happens when `deployment(id: $id)` is not passed a valid ID, so it seemed reasonable that any request failing should be a full-page failure.